### ISSUE TITLE
use of sigmod and F.binary_cross_entropy_with_logits function

### ISSUE THREAD

**Issue by XIXIYOUNG2018** [Lightning-AI/lightning#18272]
### Outline & Motivation

In lightning/examples/pytorch/domain_templates/computer_vision_fine_tuning.py  file, 
class TransferLearningModel(LightningModule):
use F.binary_cross_entropy_with_logits  as the loss function, however the output of self.fc  layer is y_logits  is pass to torch.sigmoid() and then use F.binary_cross_entropy_with_logits to for loss calculation. 
pytorch.org example F.binary_cross_entropy_with_logits  similar to torch.nn.BCEWithLogitsLoss() which combines a Sigmoid layer and the BCELoss in one single class.
In this case, the use of sigmod before  F.binary_cross_entropy_with_logits  will lead to all probability are greater than 0.5.
Such model can't distinguish negative sample from positive sample.

cc @borda @justusschock @awaelchli

**Comment by awaelchli** [comment]
That's correct @XIXIYOUNG2018, thanks for checking.
Would you like to send an update with the fix?

**Comment by mostafahadian** [comment]
It seems that it is already correct and there's no need to fix it. Because self.loss() is directly applied to y_logits, and y_logits is the direct output of self.fc():
```
    def training_step(self, batch, batch_idx):
        # 1. Forward pass:
        x, y = batch
        y_logits = self.forward(x)
        y_scores = torch.sigmoid(y_logits)
        y_true = y.view((-1, 1)).type_as(x)

        # 2. Compute loss
        train_loss = self.loss(y_logits, y_true)

        # 3. Compute accuracy:
        self.log("train_acc", self.train_acc(y_scores, y_true.int()), prog_bar=True)

        return train_loss

        # 2. Compute loss
        train_loss = self.loss(y_logits, y_true)
```
Reference:
[computer_vision_fine_tuning.py#L227](https://github.com/Lightning-AI/lightning/blob/master/examples/pytorch/domain_templates/computer_vision_fine_tuning.py#L227)

**Comment by awaelchli** [comment]
@XIXIYOUNG2018 our eyes got tricked! Thanks @mostafahadian
